packages/core/src/fhir-deduplication/deduplicate-fhir.ts

```diff
@@ -42,3 +38,22 @@ export function deduplicateFhir(fhirBundle: Bundle<Resource>): Bundle<Resource>
 
   return deduplicatedBundle;
 }
+
+/**
+ * Finds and updates references to the deduplicated resources
+ */
+function replaceResourceReferences(
+  resourceArrays: ExtractedFhirTypes,
+  idMap: Map<string, string[]>,
+  resourceType: string
+): ExtractedFhirTypes {
+  let updatedArrays = JSON.stringify(resourceArrays);
+  for (const [masterId, otherIds] of idMap.entries()) {
+    for (const id of otherIds) {
+      const regex = new RegExp(`${resourceType}/${id}`, "g");
+      updatedArrays = updatedArrays.replace(regex, masterId);
```

Comment:
this function cant accomodate replacing Resource References for more than one resourceType at a time. We might want to store Condition/id in the idMap itself and that way we can concatenate lists across different deduplication outputs and just do one big replace.

Reply:
good point. Ok lets leave it for now then